Xamarin.Forms.Core/AnimationExtensions.cs

```diff
@@ -58,7 +58,7 @@ public static bool AbortAnimation(this IAnimatable self, string handle)
 
 			if (Device.IsInvokeRequired)
 			{
-				Device.BeginInvokeOnMainThread(abort);
+				Device.BeginInvokeOnMainThread(abort, ((BindableObject)self).WindowId);
```

Comment:
Since there's already an overload of BeginInvokeOnMainThread which takes a BindableObject, why not use that instead of getting and passing the WindowId?

Also, this needs to use an `as` and a null check, not a direct cast; there's no guarantee that an `IAnimatable` is a `BindableObject` (`IAnimatable` is public, so someone may have implemented their own). I'd suggest an extension method on `IAnimatable` to handle all this.

Reply:
We've removed Cast and added a better check.